internal/controller/platform/standalone/nimservice.go

```diff
@@ -349,6 +349,12 @@ func (r *NIMServiceReconciler) reconcileNIMService(ctx context.Context, nimServi
 	var conType, failedCon string
 	var renderObj client.Object
 
+	gpuCountPerPod, err := shared.GetGPUCountPerPod(ctx, r.GetClient(), nimService)
```

Comment:
Would this function need to be invoked each time we reconcile a NIMService object?
Can we cache the GPU count per pod for DRA, so we can avoid repeatly calling this function.

An idea would be annotation to the NIMService object for GPUCountPerPod, and only run this function if the annotation is not present.

Reply:
Fixed to use annotation